pkg/api/types.go

```diff
@@ -1003,6 +1003,16 @@ type NodeSpec struct {
 
 	// Unschedulable controls node schedulability of new pods. By default node is schedulable.
 	Unschedulable bool `json:"unschedulable,omitempty"`
+
+	// Certificate is the public x.509 certificate tied to the node contained
+	// in a PEM-encoded byte array. It is presented by the node when it asks to
+	// join the cluster and can be refreshed if the certificate on the node is
+	// rotated. The node certificate is checked when initiating a connection to
+	// the node (to verify that the connection isn't being intercepted) and is
+	// presented by the node when it connects to the master (as a client
+	// certificate) to establish that the connection is coming from a trusted
+	// and identifiable source.
+	Certificate []byte `json:certificate,omitempty"`
 }
```

Comment:
For static clustering, the node will present a certificate that the master can verify because it's signed by the same CA. The master could do this on each api call and not worry about whether the certificate for a node changed over time, as long as it was signed by the appropriate CA. 

Down the road when we do dynamic clustering, the node will generate a self signed certificate and ask the master to sign it (as part of registering itself with the cluster). The master will want to verify that each time a node makes an api request it presents a certificate matching the one it originally provided when the master allowed it to join the cluster. So the master needs to hold on to that original cert. 

> the nodeSpec was created by a trusted means (cloudprovider or admin manually adding node).

As we move towards nodes explicitly joining the cluster (by way of a join request), there won't be any need for the cloud provider to create a node spec (and I plan to rip this out). 

How do you envision an admin manually adding a node? With static clustering, I see it being done by taking a copy of the cert one one node, copying it to node2, and starting a kubelet pointed at the master. With dynamic clustering, you should just be able to point a kubelet at the bootstrap endpoint (where it discovers the master and the master's cert) and then the master will decide if it should accept/reject the node or punt to the admin.